### DISCUSSION TITLE
#lang coalton? instead of coalton-top-level

### DISCUSSION THREAD

**Discussion by Inc0n** [coalton-lang/coalton#766] [Ideas and Proposals]
Would it be a good idea to have something similar to `racket` where "#lang" can be used to changed between DSLs in common lisp? Could this be beneficial to `coalton`?

Well, I think the scope of this question is probably not just limited to coalton.

**Comment by stylewarning** [comment]
There is ongoing work to allow Coalton files to be written, similar to what you suggest. Biggest roadblock is making it practical and integrated with e.g., SLIME.
